src/crimson/os/seastore/onode_manager/staged-fltree/tree.h

```diff
@@ -280,9 +280,9 @@ class Btree {
   eagain_ifuture<std::size_t> erase(Transaction& t, const ghobject_t& obj) {
     return seastar::do_with(
       full_key_t<KeyT::HOBJ>(obj),
-      [this, &t](auto& key) -> eagain_ifuture<std::size_t> {
-        return get_root(t).si_then([this, &t, &key](auto root) {
-          return root->erase(get_context(t), key, std::move(root));
+      [this, &t, &obj](auto& key) -> eagain_ifuture<std::size_t> {
+        return get_root(t).si_then([this, &t, &key, &obj](auto root) {
```

Comment:
Ditto, `obj` is not supposed to have extented lifecycle beyond async operations.